### PR TITLE
IDE-2144 Add new OSGI Module Wizard second page

### PR DESCRIPTION
**Pull request by simonjhy**

### CODE REVIEW

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java [hunk 1]
```diff
@@ -82,52 +90,173 @@ public synchronized ILiferayProject provide( Object adaptable )
...
+
+                if( location != null && location.segmentCount() > 0 )
+                {
+                    if( !lastSegment.equals( projectName ) )
+                    {
+                        projecLocation = location.append( projectName );
+                    }
+                }
+
+                ProjectImportConfiguration configuration = new ProjectImportConfiguration();
```
Comment (gamerson) [inline gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java]: Coordinate with Andy.  he has developed a workspace wizard that imports a gradle project.  He created a GradleUtil for doing this import.  Go ahead and add the util method to GradleUtil as a part of this pull request.

File: gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java [hunk 2]
```diff
@@ -82,52 +90,173 @@ public synchronized ILiferayProject provide( Object adaptable )
...
+                configuration.setWorkingSets( new ArrayList<String>() );
+                SynchronizeGradleProjectJob synchronizeGradleProjectJob =
+                    new SynchronizeGradleProjectJob(
+                        configuration.toFixedAttributes(), configuration.getWorkingSets().getValue(),
+                        AsyncHandler.NO_OP );
+
+                final IPath finalClassPath =
+                    getClassFilePath( projectName, className, packageName, projectTemplateName, projecLocation );
+
+                final File finalClassFile = finalClassPath.toFile();
```
Comment (gamerson) [inline gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java]: This should be using "IFile" not "File"

File: gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java [hunk 3]
```diff
@@ -82,52 +90,173 @@ public synchronized ILiferayProject provide( Object adaptable )
...
+        {
+            packageNamePath = packageNamePath.append( pack );
+        }
+
+        final IPath finalClassPath = packageNamePath.append( finalClassName );
+
+        return finalClassPath;
+    }
+
+    private String checkForErrors( String[] lines )
```
Comment (gamerson) [inline gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java]: Add this method to BladeCLI.  the BladeCLI class should know how to parse the output of the blade command to search for "errors"

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ComponentNameDefaultValueService.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        
+        final String projectTemplate = op().getProjectTemplateName().content( true );
+        
+        final String pName = projectName.replaceAll( "-", "" );
+        
+        final String finalProjectName = pName.replaceAll( "\\.", "" );
+        
+        final StringBuffer componentNameBuffer = new StringBuffer(finalProjectName);
+        
+        if( projectTemplate.equals( "servicebuilder" ) )
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ComponentNameDefaultValueService.java]: Need to make this more generic.  In the future we may have 50 templates.  So this wont scale.

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ModuleProjectNameValidationService.java [hunk 5]
```diff
@@ -79,7 +79,7 @@ else if( isInvalidProjectName( op ) )
             {
                 retval = Status.createErrorStatus( "A project with that name already exists." );
             }
-            else if( isMavenProject( op ) && ! isValidMavenProjectName( currentProjectName ) )
+            else if( isGradleProject( op ) && ! isValidProjectName( currentProjectName ) )
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ModuleProjectNameValidationService.java]: Don't add "gradle" code in project.core.  This validation should be delegated to ProjectProvider interface.

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/NewLiferayModuleProjectOpMethods.java [hunk 6]
```diff
@@ -30,11 +33,506 @@
...
+        "org.osgi.service.permissionadmin.PermissionAdmin",
+        "org.osgi.service.provisioning.ProvisioningService",
+        "org.osgi.service.startlevel.StartLevel",
+        "org.osgi.service.url.URLStreamHandlerService",
+        "org.osgi.service.useradmin.UserAdmin",
+        "org.springframework.context.ApplicationContext" };
+    
+    public static String[] getServices()
+    {
+        //TODO
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/NewLiferayModuleProjectOpMethods.java]: Add a new JIRA ticket for making this dynamic.  And remove this TODO.

File: tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/modules/NewLiferayModuleProjectWizard.sdef [hunk 7]
```diff
@@ -62,5 +62,44 @@
             <image>NewLiferayModuleProjectWizard.png</image>
         </page>
         <description>Create a new project configured as a Liferay plugin</description>
+        <page>
+            <id>LiferayServiceModuleSettingWizardPage</id>
+            <label>Liferay Service Or ServiceWrapper Module  Project</label>
+            <description>Set Service or ServiceWrapper Advance setting.</description>
+            <visible-when>${ ProjectTemplateName == 'servicewrapper' || ProjectTemplateName == 'service' }</visible-when>
+            <content>
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/modules/NewLiferayModuleProjectWizard.sdef]: Fix whitespace issues.

File: tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/modules/NewLiferayModuleProjectWizard.sdef [hunk 8]
```diff
@@ -62,5 +62,44 @@
...
+				<property-editor>PackageName</property-editor>
+            	<property-editor>ComponentName</property-editor>
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/modules/NewLiferayModuleProjectWizard.sdef]: Make the componentName property editor have the initial focus.

**Comment by gamerson** [comment]
Please resend, thanks! <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java (hunk 2 above)
Comment (simonjhy) [inline gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java]: why this we need to use IFile? This action will be done before the project import to current workspace, so its not a workspace project until now.

File: gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java (hunk 1 above)
Comment (simonjhy) [inline gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java]: Hi greg, I disscussed with andy that he will change this together after my code merged. Is that ok?

File: tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ModuleProjectNameValidationService.java (hunk 5 above)
Comment (simonjhy) [inline tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ModuleProjectNameValidationService.java]: I checked this code and don't need to check this while user create a new module project.

**Pull request closed without merging**